File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnFrameQueryFromClassId.java
package org.sqlunet.verbnet.sql;

import android.database.sqlite.SQLiteDatabase;

import org.sqlunet.sql.DBQuery;

/**
 * VerbNet frame query
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class VnFrameQueryFromClassId extends DBQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.VerbNetFramesQueryFromClassId;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param classId    target classId
	 */
	@SuppressWarnings("boxing")
	public VnFrameQueryFromClassId(final SQLiteDatabase connection, final long classId)
	{
		super(connection, VnFrameQueryFromClassId.QUERY);
		setParams(classId);
	}

	/**
	 * Get the frame id from the result set
	 *
	 * @return the frame id from the result set
	 */
	@SuppressWarnings("unused")
	public long getFrameId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(0);
	}

	/**
	 * Get the frame number from the result set
	 *
	 * @return the frame number from the result set
	 */
	public String getNumber()
	{
		assert this.cursor != null;
		return this.cursor.getString(1);
	}

	/**
	 * Get the frame xtag from the result set
	 *
	 * @return the frame xtag from the result set
	 */
	public String getXTag()
	{
		assert this.cursor != null;
		return this.cursor.getString(2);
	}

	/**
	 * Get the frame major description from the result set
	 *
	 * @return the frame major description from the result set
	 */
	public String getDescription1()
	{
		assert this.cursor != null;
		return this.cursor.getString(3);
	}

	/**
	 * Get the frame minor description from the result set
	 *
	 * @return the frame minor description from the result set
	 */
	public String getDescription2()
	{
		assert this.cursor != null;
		return this.cursor.getString(4);
	}

	/**
	 * Get the frame syntax from the result set
	 *
	 * @return the frame syntax from the result set
	 */
	public String getSyntax()
	{
		assert this.cursor != null;
		return this.cursor.getString(5);
	}

	/**
	 * Get the frame semantics from the result set
	 *
	 * @return the frame semantics from the result set
	 */
	public String getSemantics()
	{
		assert this.cursor != null;
		return this.cursor.getString(6);
	}

	/**
	 * Get the frame example from the result set
	 *
	 * @return the frame example from the result set
	 */
	public String getExamples()
	{
		assert this.cursor != null;
		return this.cursor.getString(7);
	}
}

File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnFrameQueryFromClassId.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.verbnet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.sql.DBQuery

/**
 * VerbNet frame query
 *
 * @param connection connection
 * @param classId    target classId
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal class VnFrameQueryFromClassId(connection: SQLiteDatabase, classId: Long) : DBQuery(connection, QUERY) {

    init {
        setParams(classId)
    }

    /**
     * Frame id from the result set
     */
    val frameId: Long
        get() = cursor!!.getLong(0)

    /**
     * Frame number from the result set
     */
    val number: String
        get() = cursor!!.getString(1)

    /**
     * Frame xtag from the result set
     */
    val xTag: String
        get() = cursor!!.getString(2)

    /**
     * Frame major description from the result set
     */
    val description1: String
        get() = cursor!!.getString(3)

    /**
     * Frame minor description from the result set
     */
    val description2: String
        get() = cursor!!.getString(4)

    /**
     * Frame syntax from the result set
     */
    val syntax: String
        get() = cursor!!.getString(5)

    /**
     * Frame semantics from the result set
     */
    val semantics: String
        get() = cursor!!.getString(6)

    /**
     * Frame example from the result set
     */
    val examples: String
        get() = cursor!!.getString(7)

    companion object {
        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.VerbNetFramesQueryFromClassId
    }
}
